packages/storage-plugin/src/symbols.ts

```diff
@@ -24,3 +30,11 @@ export interface NgxsStoragePluginOptions {
 }
 
 export const NGXS_STORAGE_PLUGIN_OPTIONS = new InjectionToken('NGXS_STORAGE_PLUGIN_OPTION');
+
+export const STORAGE_ENGINE = new InjectionToken('STORAGE_ENGINE');
+
+export interface StorageEngine {
+  getItem(key): any;
+  setItem(key, val): void;
+  removeItem(key): void;
+}
```

Comment:
we can add `length`, `key` and `clear` as well to make it fully compatible.

Reply:
Everything else looks great. We just need to test the non-default option, for example sessionStorage.

Reply:
Ill make sure it is tested. Just wanted your input before I went to far